libsrc/telestrat/ch376.s

```diff
@@ -0,0 +1,344 @@
+    bne     loop
+end:	
+    sta     CH376_DATA
+    rts
+.endproc 
+	
+; char _ch376_file_open();
+.proc _ch376_file_open
+    lda     #CH376_FILE_OPEN ; $32
+    sta     CH376_COMMAND
+    jsr     _ch376_wait_response
+    rts
+.endproc 
+	
+.proc _ch376_get_file_size
+    lda     #CH376_GET_FILE_SIZE
+    sta     CH376_COMMAND
+    lda     #$68
+    sta     CH376_DATA
+    ; store file length 32 bits
+    lda     CH376_DATA
+    sta     tmp1
+    lda     CH376_DATA
+    sta     tmp1+1
+    lda     CH376_DATA
+    sta     tmp2
+    lda     CH376_DATA
+    sta     tmp2+1
+    rts
+.endproc 
+
+; void ch376_reset();
+.proc _ch376_reset
+    lda     #CH376_RESET_ALL ; 5 
+    sta     CH376_COMMAND
+    ; waiting
+    ldy     #0
+    ldx     #0
+loop:
+    nop
+    inx
+    bne     loop
+    iny
+    bne     loop
+    rts
+.endproc 
+
+; char  ch376_check_exist(char value);
+	
+.proc _ch376_check_exist
+    sta     tmp1
+    lda     #CH376_CHECK_EXIST ; 
+    sta     CH376_COMMAND
+    lda     tmp1
+    sta     CH376_DATA
+    lda     CH376_DATA
+    rts
+.endproc 
+	
+; char 	ch376_ic_get_version(void)
+.proc _ch376_ic_get_version
+    lda     #CH376_GET_IC_VER
+    sta     CH376_COMMAND
+    ldx     #0
+    lda     CH376_DATA
+    rts
+.endproc 
+
+; void ch376_set_usb_mode(char mode)
+.proc _ch376_set_usb_mode
+    ldx     #CH376_SET_USB_MODE ; $15
+    stx     CH376_COMMAND
+    sta     CH376_DATA
+    rts
+.endproc 
+		
+; 	void ch376_set_bytes_write(int value);
+.proc _ch376_set_bytes_write
+    ldy     #CH376_BYTE_WRITE
+    sty     CH376_COMMAND
+    sta     CH376_DATA
+    stx     CH376_DATA
+    lda     #0
+    sta     CH376_DATA
+    sta     CH376_DATA
+    jsr     _ch376_wait_response
+    rts
+.endproc 	
+	
+.proc _ch376_set_bytes_read
+    ldy     #CH376_BYTE_READ
+    sty     CH376_COMMAND
+    ; Storing 32 bits value
+    sta     CH376_DATA
+    stx     CH376_DATA
+    lda     #0
+    sta     CH376_DATA
+    sta     CH376_DATA
+    jsr     _ch376_wait_response
+	rts
+.endproc 		
+
+; char 	ch376_disk_mount();
+.proc _ch376_disk_mount
+    lda     #CH376_DISK_MOUNT ; $31
+    sta     CH376_COMMAND
+    jsr     _ch376_wait_response
+    ; if we read data value, we have then length of the volume name
+    ldx     #0
+    rts	
+.endproc 
+
+
+; char 	ch376_wait_response();
+.proc _ch376_wait_response
+;  return 1 if usb controller does not respond
+; else A contains answer of the controller
+    ldy     #$FF ; We have to wait 35 ms, but well, this loop is broken when controler is OK
+loop3:
+    ldx     #$FF ; don't decrease this counter. Because ch376 won't respond if there is a lower value
+loop:
+    lda     CH376_COMMAND
+    and     #%10000000
+    cmp     #128
+    bne     no_error
+    dex
+    bne     loop
+    dey
+    bne     loop3
+    ; error is here
+    rts
+no_error:
+    lda     #CH376_GET_STATUS
+    sta     CH376_COMMAND
+    lda     CH376_DATA
+    rts
+.endproc 
+
+.proc _ch376_fread
```

Comment:
It's a code to test loading from usbkey because i did a tool with cc65 to test all the functionnalities of the chip. I removed ch376.s file in order to link with my tool, instead of sharing the source code in cc65